src/test/java/seedu/address/model/person/DonorTest.java

```diff
@@ -82,4 +84,19 @@ public void equals() {
         Person editedAlice = new PersonBuilder(ALICE).withType("doctor").build();
         assertFalse(editedAlice.equals(editedJohn));
     }
+
+    @Test
+    public void toStringTest() {
+        Donor john = new DonorBuilder(JOHN).build();
+        assertEquals(john.toString().trim() , "John Person Type: donor Nric: T1312123P Phone: 81230942 Age: 60");
+    }
+
+    @Test
+    public void hashCodeTest() {
+        Donor john = new DonorBuilder(JOHN).build();
+
+        assertEquals(john.hashCode(), new DonorBuilder(JOHN).build().hashCode());
+        assertNotEquals(john.hashCode(),
+                new DonorBuilder(JOHN).withPhone(VALID_PHONE_BOB).build().hashCode());
+    }
```

Comment:
I propose renaming the typical persons to follow this format: `PERSONTYPE_PERSONNAME` e.g. `DONOR_JOHN`. Will make an issue for discussion, no action needed for this PR.